Scripts/Extensions/RendererAnimationExtensions.cs

```diff
@@ -6,31 +6,94 @@ namespace DUCK.Tween.Extensions
 		/// <param name="easingFunction">The easing function that will be used to interpolate with</param>
 		/// <returns>A DelegateAnimation which contains the created RendererColorFadeAnimation</returns>
-		public static DelegateAnimation<RendererColorFadeAnimation> ColorFadeTo(this Renderer renderer, Color to, float duration = 1f, Func<float, float> easingFunction = null)
+		public static DelegateAnimation<ColorShaderPropertyAnimation> ColorFadeTo(this Renderer renderer, Color to, float duration = 1f, Func<float, float> easingFunction = null)
 		{
-			var animation = new DelegateAnimation<RendererColorFadeAnimation>(() =>
-				new RendererColorFadeAnimation(renderer, renderer.material.color, to, duration, easingFunction));
+			var animation = new DelegateAnimation<ColorShaderPropertyAnimation>(() =>
+				new ColorShaderPropertyAnimation(renderer, renderer.material.color, to, duration, easingFunction));
+			return animation;
+		}
+
+		/// <summary>
+		/// Creates a new ColorShaderPropertyAnimation using this renderer.
+		/// </summary>
+		/// <param name="renderer">The renderer that will be the target of the animation</param>
+		/// <param name="shaderProperty">The name of the target shader property</param>
+		/// <param name="from">The start alpha value of the fade</param>
+		/// <param name="to">The end alpha value of the fade</param>
+		/// <param name="duration">The duration of the animation in seconds, defaults to 1f</param>
+		/// <param name="easingFunction">The easing function that will be used to interpolate with</param>
+		/// <returns>The RendererFadeAnimation created</returns>
+		public static ColorShaderPropertyAnimation ColorFade(this Renderer renderer, string shaderProperty, Color from, Color to, float duration = 1f, Func<float, float> easingFunction = null)
+		{
+			return new ColorShaderPropertyAnimation(renderer, shaderProperty, from, to, duration, easingFunction);
+		}
+
+		/// <summary>
+		/// Creates a new ColorShaderPropertyAnimation (in a DelegateAnimation) using this renderer
+		/// </summary>
+		/// <param name="renderer">The renderer that will be the target of the animation</param>
+		/// <param name="shaderProperty">The name of the target shader property</param>
+		/// <param name="to">The end color of the fade</param>
+		/// <param name="duration">The duration of the animation in seconds, defaults to 1f</param>
+		/// <param name="easingFunction">The easing function that will be used to interpolate with</param>
+		/// <returns>A DelegateAnimation which contains the created RendererColorFadeAnimation</returns>
+		public static DelegateAnimation<ColorShaderPropertyAnimation> ColorFadeTo(this Renderer renderer, string shaderProperty, Color to, float duration = 1f, Func<float, float> easingFunction = null)
+		{
+			var shaderPropertyID = Shader.PropertyToID(shaderProperty);
+			var animation = new DelegateAnimation<ColorShaderPropertyAnimation>(() =>
```

Comment:
You can just do this:
`return renderer.ColorFadeTo(shaderPropertyID, renderer.material.GetColor(shaderPropertyID), to, duration, easingFunction));`